as/thrift.js

```diff
@@ -66,17 +66,13 @@ function TChannelAsThrift(opts) {
         'channel must be provided with isHealthy');
 
     if (self.isHealthy) {
-        fs.readFile(path.join(__dirname, 'meta.thrift'), 'utf8', registerHealthCheck);
+        var thriftSource = fs.readFileSync(
```

Comment:
Agreed it's bad. I wanted to fix the race condition. I can also fix race condition with `setTimeout` in tests.

The same race condition applies to anyone wanting to test their health check. Not sure if there is a good fix without breaking back compat.